tests/test_sails.py

```diff
@@ -0,0 +1,231 @@
+  FAIL:     Status code is not 400 or the number of test01 is not 1.
+  """
+  # Add record
+  collection.insert_one(query)
+  
+  # Act
+  r = requests.post('http://capstonedd.cs.pdx.edu:1337/users', data=dataquery)
+
+  # Assert and redo
+  c = collection.count(findq)
+  if c == 1:
+    collection.delete_one(findq)
+    assert r.status_code == 400
+  else:
+    if c > 1:
+      collection.delete_many(findq)
+    raise AssertionError
+  
+def test_bad_post2_returns_400_code():
+  """
+  SUMMARY:  Test that invalid POST method returns a 400 code ('bad request')
+  METHOD:   (1) Create a improper payload (no firstName).
+            (2) Send a POST request with improper post body.
+  FAIL:     Status code is not 400 or number of test02 is not zero.
+  """
+  # Clean
+  c = collection.count(testq)
+  if c > 0:
+    collection.delete_many(testq)
+  
+  # Arrange
+  payload = {
+    'lastName' : 'test02',
+    'email' : 'test02@pdxcapstone.com',
+    'type' : 'buyer'
+  }
+  
+  # Act
+  r = requests.post('http://capstonedd.cs.pdx.edu:1337/users', data=json.dumps(payload))
+
+  # Assert
+  c = collection.count(testq)
+  if c > 0:
+    collection.delete_many(testq)
+    raise AssertionError
+  else:
+    assert r.status_code == 400
+  
+def test_bad_post3_returns_400_code():
+  """
+  SUMMARY:  Test that invalid POST method returns a 400 code ('bad request')
+  METHOD:   (1) Create a improper payload (empty firstName).
+            (2) Send a POST request with improper post body.
+  FAIL:     Status code is not 400 or number of test02 is not zero.
+  """
+  
+  # Clean
+  c = collection.count(testq)
+  if c > 0:
+    collection.delete_many(testq)  
+
+  # Arrange
+  payload = {
+    'firstName' : '',
+    'lastName' : 'test02',
+    'email' : 'test02@pdxcapstone.com',
+    'type' : 'buyer'
+  }
+  
+  # Act
+  r = requests.post('http://capstonedd.cs.pdx.edu:1337/users', data=json.dumps(payload))
+
+  # Assert
+  c = collection.count(testq)
+  if c > 0:
+    collection.delete_many(testq)
+    raise AssertionError
+  else:
+    assert r.status_code == 400
+  
+def test_bad_delete_returns_400_code():
+  """
+  SUMMARY:  Test that invalid DELETE method returns a 400 code ('bad request')
+  METHOD:   (1) Create a improper payload (No records).
+            (2) Send a POST request with improper post body.
+  FAIL:     Status code is not 400.
+  """
+  
+  # Clean
+  c = collection.count(testq)
+  if c > 0:
+    collection.delete_many(testq)
+  
+  # Arrange
+  payload = {
+    'lastName' : 'test02',
+  }
+  
+  # Act
+  r = requests.post('http://capstonedd.cs.pdx.edu:1337/users', data=json.dumps(payload))
+
+  # Assert
+  assert r.status_code == 400
+  
+  
+def test_delete_returns_200_code():
+  """
+  SUMMARY:  Test that proper DELETE method returns a 200 code ('ok')
+  METHOD:   Send a DELETE request with proper post body.
+  FAIL:     Status code is not 200 or the record has not been deleted.
+  """
+  
+  # Add record
+  collection.insert_one(query)
```

Comment:
Let's keep the arrange, act, assert pattern here as well. Because we're setting the scene for a action, this should be under the arrange header, not "add record".